Events/docs (#9100)
hashicorp/nomad: website/pages/api-docs/events.mdx

---
layout: api
page_title: Events - HTTP API
sidebar_title: Events
description: |-
  The /event endpoints interact with the local Nomad agent to interact with

>   members and servers.

Reviewer: How about something like, "The /event endpoints are used to query for and interact with Nomad events"

Author: :grimacing: woops, that's a bad copy paste :sweat_smile:

---

# Events HTTP API

The `/event` endpoints are used to query for recent events.

## Event Stream

> This endpoint streams a servers backlog off events as we as new events as they occur.

Reviewer: ```suggestion
This endpoint streams a servers backlog of events as well as new events as they occur.
```

The stream will be kept alive until the connection is closed.

| Method | Path               | Produces           |
| ------ | ------------------ | ------------------ |
| `GET`  | `/v1/event/stream` | `application/json` |

> The Table below shows this endpoint's [required ACLs](/api-docs#acls). Due to

Reviewer: ```suggestion
The table below shows this endpoint's [required ACLs](/api-docs#acls). Due to
```

> the nature of this endpoint different topics require different policies.

Reviewer: ```suggestion
the nature of this endpoint individual topics require specific policies.
```

| Topic        | ACL Required         |
| ------------ | -------------------- |
| `*`          | `management`         |
| `Job`        | `namespace:read-job` |
| `Alloc`      | `namespace:read-job` |
| `Deployment` | `namespace:read-job` |
| `Eval`       | `namespace:read-job` |
| `Node`       | `node:read`          |

### Parameters

- `index` `(int: 0)` - Specifies the index to start streaming events from. If

>   the requeste index is no longer in the buffer the stream will start at the

Reviewer: ```suggestion
  the requested index is no longer in the buffer the stream will start at the
```

>   nearest index.

Reviewer: ```suggestion
  next available index.
```

> - `topic` `(topic:filter_key: "*:*")` - Specifies a topic to subscribe to and

Reviewer: ```suggestion
- `topic` `(topic:filter_key: "*:*")` - Specifies the set of topics to subscribe to and
```

>   filter on. By default all topics are subscribed to. Multiple topics should be

Reviewer: ```suggestion
  filter on. The default is to subscribe to all topics. Multiple topics may be
```

  specified by passing multiple `topic` parameters. A valid topic parameter
  includes a `topic` type and an optional `filter_key` separated by a colon
  `:`. As an example `?topic=Deployment:redis` would subscribe to all

>   `Deployment` events for a job redis. To only subscribe to `Node` events a

Reviewer: Can there be multiple filters per topic?

Author: yes will add to this to reflect
